external-crates/move/crates/move-package/tests/test_lock_file.rs

```diff
@@ -61,6 +65,68 @@ fn discard() {
     assert!(!pkg.path().join("Move.lock").is_file());
 }
 
+#[test]
+fn update_lock_file_toolchain_version() {
+    let pkg = create_test_package().unwrap();
+    let lock_path = pkg.path().join("Move.lock");
+
+    let lock = LockFile::new(
+        pkg.path().to_path_buf(),
+        /* manifest_digest */ "42".to_string(),
+        /* deps_digest */ "7".to_string(),
+    )
+    .unwrap();
+    lock.commit(&lock_path).unwrap();
+
+    let mut build_config: BuildConfig = Default::default();
+    build_config.default_flavor = Some(Flavor::Sui);
+    build_config.default_edition = Some(Edition {
+        edition: Symbol::from("2024"),
+        release: Some(Symbol::from("1")),
+    });
+    build_config.lock_file = Some(lock_path.clone());
+    let _ = build_config.update_lock_file_toolchain_version("0.0.1".into());
+
+    let lock_contents = {
+        let mut lock_file = File::open(lock_path).unwrap();
+        let mut buf = String::new();
+        lock_file.read_to_string(&mut buf).unwrap();
+        buf
+    };
+
+    let mut toml = lock_contents.parse::<toml_edit::Document>().unwrap();
+    let move_table = toml["move"].as_table_mut().unwrap();
+    let compiler_flags = move_table["compiler-flags"].as_table_mut().unwrap();
+    compiler_flags["lock_file"] = toml_edit::Item::Value("...".into());
+
+    let expected = expect![[r##"
+        # @generated by Move, please check-in and do not edit manually.
+
+        [move]
+        version = 0
+        manifest_digest = "42"
+        deps_digest = "7"
+        compiler-version = "0.0.1"
+
+        [move.compiler-flags]
+        default_edition = "2024.1"
+        default_flavor = "sui"
+        deps_as_root = false
+        dev_mode = false
+        fetch_deps_only = false
+        force_recompilation = false
+        generate_docs = false
+        lock_file = "..."
+        silence_warnings = false
+        skip_fetch_latest_git_deps = false
+        test_mode = false
+        warnings_are_errors = false
```

Comment:
We shouldn't include the full `BuildConfig`, most of the stuff in here can change without changing the lock file, like:

- `deps_as_root`
- `dev_mode`
- `fetch_deps_only`
- `force_recompilation`
- `generate_docs`
- `lock_file`
- `silence_warnings`
- `skip_fetch_latest_git_deps`
- `test_mode`
- `warnings_are_errors`
- `additional_named_addresses`

In fact, the only ones that we should include are probably the edition and flavor.

Reply:
Ah, fair! Running it back in https://github.com/MystenLabs/sui/pull/14973